pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -373,9 +401,27 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 			}
 		}
 
-		bundleUpdated = updateBundleCondition(st.bundle, &inProgressCond) || bundleUpdated
-		bundleUpdated = updateBundleCondition(st.bundle, &readyCond) || bundleUpdated
-		bundleUpdated = updateBundleCondition(st.bundle, &errorCond) || bundleUpdated
+		inProgressUpdated := updateBundleCondition(st.bundle, &inProgressCond) || bundleUpdated
+		readyUpdated := updateBundleCondition(st.bundle, &readyCond) || bundleUpdated
+		errorUpdated := updateBundleCondition(st.bundle, &errorCond) || bundleUpdated
+
+		if inProgressUpdated && inProgressCond.Status == smith_v1.ConditionTrue {
+			st.bundleTransitionCounter.
+				WithLabelValues(st.bundle.GetNamespace(), st.bundle.GetName(), string(inProgressCond.Type), inProgressCond.Reason).
+				Inc()
+		}
+		if readyUpdated && readyCond.Status == smith_v1.ConditionTrue {
+			st.bundleTransitionCounter.
+				WithLabelValues(st.bundle.GetNamespace(), st.bundle.GetName(), string(readyCond.Type), readyCond.Reason).
+				Inc()
+		}
+		if errorUpdated && errorCond.Status == smith_v1.ConditionTrue {
+			st.bundleTransitionCounter.
+				WithLabelValues(st.bundle.GetNamespace(), st.bundle.GetName(), string(errorCond.Type), errorCond.Reason).
+				Inc()
+		}
```

Comment:
I think all these `if`s can be moved to `updateBundleCondition()`, right? Same with `updateResourceCondition()`.